Signals: Alpha Vantage Example Script (#79)
numerai/example-scripts: signals/alphavantage/README.md

This script uses data from [AlphaVantage](https://www.alphavantage.co/) using raw queries which do not require any extra library.
```
pip install numerapi tqdm
```
`numerapi` and `tqdm` (to check progress of downloads) are needed.

`key = "<ALPHAVANTAGE API KEY>"` **this needs to be set before running the script.**

> You can generate a premium key by subscribing to [premium membership](https://www.alphavantage.co/premium/). This script was created with the lowest 75 calls/minute plan so it would work. This takes around 13 mins to load all US stocks.

Reviewer: ```suggestion
AlphaVantage has a free key with a limit of 5 API calls per minute. For a limit of 75 calls per minute, you can pay $50 USD per month for a [premium key](https://www.alphavantage.co/premium/). This script was created for use with the 75 calls per minute plan so it would work. This takes around 13 mins to load all US stocks.
```

> Alpha Vantage offers various typed of data including Stocks, cryptocurrencies, and technical indicators. [Documentation](https://www.alphavantage.co/documentation/).

Reviewer: ```suggestion
Alpha Vantage offers various types of data including stocks, cryptocurrencies, and technical indicators. [Documentation](https://www.alphavantage.co/documentation/).
```

### Data Loading

> Since the API has some rate limits of fixed calls/minute, this script provides two functions for loading the data.

Reviewer: ```suggestion
Since the API has a rate limit of calls per minute, this script provides two functions for loading the data.
```

- Sequential 
    + This may take over an hour to load all available tickers.
    + Safe and loading function can be stopped.
    + `full_data = load_data(tickers, "full_data.csv", threads=False)`

- Parallel
    + uses threads from Python's `concurrent` module.
    + much faster loading (~13 mins for all US tckers)
    + execution is not stoppable.
    + `full_data = load_data(tickers, "full_data.csv", threads=True)`
    + Thanks to **[Jordi Villar](https://twitter.com/jrdi)** for supercharging the parallel execution and bringing it to ~13 mins.

Curently available tickers from the universe using yahoo tickers are in `tickers.csv.txt` file. 

`tickers = pd.read_csv("tickers.csv.txt")`

> 

Reviewer: Should these lines be removed? Looks like you removed the file named `tickers.csv.txt`

Author: removed

### Feature Generation

> - Since Alpha vantage has differnet types of data resolutions, this script uses **weekly data** (OHLC) to generate features.

Reviewer: ```suggestion
- Since Alpha vantage has different types of data resolutions, this script uses **weekly data** (OHLC) to generate features.
```

- Some features are `simple moveing average` and `exponential moving average` of periods `[2, 5, 21, 50, 200]`.

> 

Reviewer: Could you add the diagnostics sidebar image you sent me in RocketChat?

Author: Added. Thank you
